org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -298,48 +314,61 @@ class Translator {
 
-            var postPayload = {
-                callback: subscriptionInfo.callbackUrl
-            }
+                var postPayload = {
+                    callback: subscriptionInfo.callbackUrl
+                }
 
-            // Secret provided for computing HMAC verification of the payload
-            // this is optional to Wink
-            if (subscriptionInfo.key) {
-                postPayload.secret = subscriptionInfo.key;
-            }
+                // Secret provided for computing HMAC verification of the payload
+                // this is optional to Wink
+                if (subscriptionInfo.key) {
+                    postPayload.secret = subscriptionInfo.key;
+                }
 
-            var postPayloadString = JSON.stringify(postPayload);
+                var postPayloadString = JSON.stringify(postPayload);
 
-            return this._makeRequest(requestPath, 'POST', postPayloadString).then((response) => {
-                // Return the expiration time for the subscription
-                return {
-                    expiration: response.data.expires_at
-                };
-            })
+                return this._makeRequest(requestPath, 'POST', postPayloadString).then((response) => {
+                    // Return the expiration time for the subscription
+                    return {
+                        expiration: response.data.expires_at
+                    };
+                })
+            });
         } 
     }
 
     /**
+     * Gets the request path for the subscription depending on whether it should subscribe to a single platform or to 
+     * device graph updates on the Wink account.
+     */
+    _getCallbackSubscriptionRequestPath(subscriptionInfo) {
+        if (subscriptionInfo.deviceType && subscriptionInfo.deviceId) {
+                // Platform level subscription
+                return Promise.resolve(`/${subscriptionInfo.deviceType}/${subscriptionInfo.deviceId}/subscriptions`);
+            }
+            else {
+                // Subscribe to device graph updates.
+                // Wink provides a .all groupd at /groups/.all which can be subscribed to for both device graph updates
+                // and ALL platform updates.  Since platform updates can be subscribed to individually, we'll use a different
+                // subscription URL for Wink device changes.
+                // If in the future, support for subscribing to all platforms at once is needed, here is how it will need to be done:
+                //      1. GET to /groups/.all
+                //      2. Retrieve the group_id from the result to get the ID associated with the .all group
+                //      3. POST subscription request to /groups/${group_id}/subscription
+                // For now, just subscribe to device updates found at the following url.
+
+                return Promise.resolve('/users/me/wink_devices/subscriptions');
+            }
+    }
```

Comment:
Nit:  indenting looks off here.  Can you format the document (alt-shift-f in vs code).